r/tests/testthat/helper-expectation.R

```diff
@@ -98,12 +103,16 @@ expect_dplyr_equal <- function(expr, # A dplyr pipeline with `input` as its star
 expect_dplyr_error <- function(expr, # A dplyr pipeline with `input` as its start
                                tbl,  # A tbl/df as reference, will make RB/Table with
                                ...) {
+  # ensure we have supplied tbl
+  tbl
+  
   expr <- rlang::enquo(expr)
   msg <- tryCatch(
     rlang::eval_tidy(expr, rlang::new_data_mask(rlang::env(input = tbl))),
     error = function (e) conditionMessage(e)
```

Comment:
```suggestion
    error = function (e) {
          msg <- conditionMessage(e)

      # The error here is of the form:
      #
      # Problem with `filter()` input `..1`.
      # x object 'b_var' not found
      # ℹ Input `..1` is `chr == b_var`.
      #
      # but what we really care about is the `x` block
      # so (temporarily) let's pull those blocks out when we find them
      if (grepl("object '.*'.not.found", msg)) {
        return(gsub("^.*(object '.*'.not found).*$", "\\1", msg))
      }
      if (grepl('could not find function ".*"', msg)) {
        return(gsub('^.*(could not find function ".*").*$', "\\1", msg))
      }
      invisible(structure(msg, class = "try-error", condition = e))
    }
```

Reply:
@jonkeane I don't fully understand why this is necessary, but as it turns out your regular expressions are already encapsulated in `i18ize_error_messages()` (https://github.com/apache/arrow/blob/master/r/R/dplyr.R#L374), so maybe that can be reused.

Also I believe you just want to return `msg` from this, not a `try-error`, at least that's the current behavior and also what the early returns you added do.

Reply:
Yeah, returning `msg` makes sense there, I dug the core of this from git history and that's what we returned when those `grepl()`s didn't find anything before — though I bet it wasn't exercised. Returning simply `msg` sounds right.